server/src/core/stat/stat-service.ts

```diff
@@ -0,0 +1,98 @@
+
+const COLLECTION = 'stats';
+
+export interface StatsArgs {
+  transactions: TransactionService
+}
+
+export class Statistics implements StatsService {
+  private db: Db;
+  private collection: Collection<Stats>;
+  private transactions: TransactionService;
+
+  constructor(db: Db, args: StatsArgs) {
+    this.db = db;
+    this.collection = this.db.collection(COLLECTION);
+    this.transactions = args.transactions;
+  }
+
+  async get(): Promise<Stats> {
+    try {
+      let stats = await this.collection.findOne({ _id: 'stats' });
+      if (!stats) {
+        stats = await this.update(); 
+      } 
+      return stats;
+    }
+    catch (e) {
+      rethrowIfAppError(e);
+      throw createDbOpFailedError(e.message);
+    }
+  }
+
+  async update(): Promise<Stats> {
+    try {
+      const res = await this.transactions.aggregate([
+        {
+          $facet: {
+            numContributorsPipeline: [
+              { $match: { type: 'donation', status: 'success'} },
+              { $group: { _id: "$to" } },
+              { $count: 'numContributors' }
+            ],
+            numBeneficiariesPipeline: [
+              { $match: { type: 'distribution', status: 'success'} },
+              { $group: { _id: "$to" } },
+              { $count: 'numBeneficiaries' }
+            ],
+            totalContributedPipeline: [
+              { $match: { type: 'donation', status: 'success' } },
+              { $group: { _id: null, totalContributed: { $sum: "$amount" } } },
+              { $project: { _id: 0, totalContributed: 1 } }
+            ],
+            totalDistributedPipeline: [
+              { $match: { type: 'distribution', status: 'success' } },
+              { $group: { _id: null, totalDistributed: { $sum: "$amount" } } },
+              { $project: { _id: 0, totalDistributed: 1 } }
+            ],
+          }
+        }
+      ]);
+
+      let numContributors = 0;
+      let numBeneficiaries = 0;
+      let totalContributed = 0;
+      let totalDistributed = 0;
+
+      if (res.length) {
+        numContributors = res[0].numContributorsPipeline.length ? res[0].numContributorsPipeline[0].numContributors : 0;
+        numBeneficiaries = res[0].numBeneficiariesPipeline.length ? res[0].numBeneficiariesPipeline[0].numBeneficiaries : 0;
+        totalContributed = res[0].totalContributedPipeline.length ? res[0].totalContributedPipeline[0].totalContributed : 0;
+        totalDistributed = res[0].totalDistributedPipeline.length ? res[0].totalDistributedPipeline[0].totalDistributed : 0;
+      }
+
+      const updatedStats = await this.collection.findOneAndUpdate(
+        { _id: 'stats' },
+        { 
+          $set: { numContributors, totalContributed, numBeneficiaries, totalDistributed },
+          $currentDate: { updatedAt: true },
+        },
+        { upsert: true, returnOriginal: false }
+      );
+
+      if (!updatedStats) throw createResourceNotFoundError(messages.ERROR_STATS_NOT_FOUND);
```

Comment:
I don't think this should be a `resourceNotFoundError` because stats should be system generated (not that `upsert` is set to true). If they don't exist, it means there's some weird system error. So this should be some serverError that should return 500, not a 404.